packages/design-system-react/src/components/button-base/ButtonBase.types.ts

```diff
@@ -0,0 +1,89 @@
+import type { ComponentProps } from 'react';
+
+import type { MakePropsOptional } from '../../types/make-props-optional';
+import type { IconName, IconProps } from '../icon';
+
+export enum ButtonBaseSize {
```

Comment:
Shouldnt this be ButtonSize?

Reply:
The enum name `ButtonBaseSize` follows our established naming pattern of `ComponentName` + `Prop` = `EnumName`. 

In this case:
- Component name: `ButtonBase`
- Prop name: `size`
- Resulting enum name: `ButtonBaseSize`

This naming convention helps maintain consistency across our component library and makes the relationship between components, props, and their associated types more explicit.

Reply:
Issue here https://github.com/MetaMask/metamask-design-system/issues/292

Reply:
are we intended for the user to import `ButtonSize` or `ButtonBaseSize`? I assume it should be `ButtonSize`. If that's the case, are you intending to map the `ButtonSize` to `ButtonBaseSize`?

Reply:
Yes see the draft of `ButtonPrimary` we are exporting `ButtonBaseSize` as `ButtonPrimarySize` and importing it for usage with `ButtonPrimary` 

https://github.com/MetaMask/metamask-design-system/pull/289/files#diff-ff7a98d6e5b8ccd27ebdec0b0c9a70b9551ef9651863121cc65b58ab08fefc90R3

https://github.com/MetaMask/metamask-design-system/pull/289/files#diff-fe3e24d7920d4e19ada30de93b6830e5439b0ff9d69da9a1402101ed4ca92a11R4

Reply:
sounds good. I just want to make sure the user doesnt need to do something like 
`<Button size={ButtonBaseSize.Lg}/>`

Reply:
Yup, good call we definitely want to align the enum with the exposed component.